packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -199,9 +203,133 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
+            matchCase: !!searchOptions.matchCase,
+            matchWholeWord: !!searchOptions.matchWholeWord,
+            limitResultCount: searchOptions.maxResults
+        });
+
+        const matches: SearchMatch[] = [];
+        results.forEach(r => {
+            if (!widget.editor.getLineContent) {
+                return [];
+            }
+            const lineText: string = widget.editor.getLineContent(r.range.start.line);
+            matches.push({
+                line: r.range.start.line,
+                character: r.range.start.character,
+                length: r.range.end.character - r.range.start.character,
+                lineText
+            });
+        });
+
+        return matches;
+    }
+
+    /**
+     * Find the list of editors which meet the filtering criteria.
+     * @param editors the list of editors to filter.
+     * @param searchOptions the search options to apply.
+     */
+    protected findMatchedEditors(editors: EditorWidget[], searchOptions: SearchInWorkspaceOptions): EditorWidget[] {
+        if (!editors.length) {
+            return [];
+        }
+
+        const ignoredPatterns = this.getExcludeGlobs(searchOptions.exclude);
+        editors = editors.filter(widget => !ignoredPatterns.some(pattern => minimatch(widget.editor.uri.toString(), pattern, { dot: true, matchBase: true })));
+
+        // Only include widgets that in `files to include`.
+        if (searchOptions.include && searchOptions.include.length > 0) {
+            const includePatterns: string[] = searchOptions.include;
+            editors = editors.filter(widget => includePatterns.some(pattern => minimatch(widget.editor.uri.toString(), pattern, { dot: true, matchBase: true })));
+        }
+
+        return editors;
+    }
+
+    /**
+     * @returns whether there are open editors.
+     */
+    hasOpenEditor(): boolean {
+        return (this.editorManager.all.length > 0);
+    }
+    /**
+     * Perform a search in all open editors.
+     * @param searchTerm the search term.
+     * @param searchOptions the search options to apply.
+     *
+     * @returns the tuple of result count, and the list of search results.
+     */
+    protected searchInOpenEditors(searchTerm: string, searchOptions: SearchInWorkspaceOptions): {
+        numberOfResults: number,
+        matches: SearchInWorkspaceResult[]
+    } {
+        // Track the number of results found.
+        let numberOfResults = 0;
+
+        const searchResults: SearchInWorkspaceResult[] = [];
+        const editors = this.findMatchedEditors(this.editorManager.all, searchOptions);
+        editors.forEach(async widget => {
+            const matches = this.findMatches(searchTerm, widget, searchOptions);
+            if (matches?.length) {
+                numberOfResults += matches.length;
+                const fileUri: string = widget.editor.uri.toString();
+                const root: string = this.workspaceService.getWorkspaceRootUri(widget.editor.uri)?.toString()!;
```

Comment:
Checking for nullish and finally forcing the value to not be interpreted as not nullish is a bit weird.

Either do a proper check or move the ending `!` to replace the first `?`.